src/fluvio.rs

```diff
@@ -0,0 +1,86 @@
+use crate::SharedFluvio;
+use crate::admin::FluvioAdminWrapper;
+use crate::consumer::PartitionConsumerWrapper;
+use crate::producer::TopicProducerWrapper;
+
+use std::sync::Arc;
+
+use log::debug;
+
+use fluvio::{Fluvio};
+
+use flv_future_aio::sync::RwLock;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::core::NjError;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+
+// simple wrapper to facilitate conversion to JS Class
+pub struct FluvioWrapper(Fluvio);
+
+impl From<Fluvio> for FluvioWrapper {
+    fn from(client: Fluvio) -> Self {
+        Self(client)
+    }
+}
+
+impl TryIntoJs for FluvioWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        debug!("converting FluvioWrapper to js");
+        let new_instance = FluvioJS::new_instance(js_env, vec![])?;
+        debug!("instance created");
+        FluvioJS::unwrap_mut(js_env, new_instance)?.set_client(self.0);
+        Ok(new_instance)
+    }
+}
+
+pub struct FluvioJS {
+    inner: Option<SharedFluvio>,
```

Comment:
I've noticed that almost all of your JS structs have inner values that are all `Option<Thing>`. Why is that? Is that some limitation of how node_bindgen works? Or is it a design choice for this particular library? I'm seeing a lot of mutation tendencies and `unwrap`'s, so I'm wondering if this is something that can be avoided.

Reply:
Yes, this is due to limitation imposed by `N-API`.   Maybe we come up better design later.